onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -1194,6 +1194,11 @@ TensorrtExecutionProvider::~TensorrtExecutionProvider() {
     }
   }
 
+  if (external_stream_) {
+    ORT_IGNORE_RETURN_VALUE(CUBLAS_CALL(cublasDestroy(external_cublas_handle_)));
```

Comment:
Can we store these in class member unique_ptrs (with custom deleter) at the time of creation so that we don't have to remember to destroy them? It looks like this code is repeated in both `~PerThreadContext` and `~TensorrtExecutionProvider`.
The same applies to [this part](https://github.com/microsoft/onnxruntime/blob/f17b6afe3c5241525c3ee1384f98dbef64bcffbc/onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc#L1207-L1211) of the code.